### ISSUE TITLE
Notification fired twice when openning the app from the notification.

### ISSUE THREAD

**Issue by sagivf** [driftyco/ionic-cloud-angular#23]
After sending a single notification.
- Opening the app fires only once. **Great!**
- Opening the app from the notification after the app was once opened only fires once.
  **This sounds reasonable to me but perhaps an extra field on the notification would be nice**.
- Opening the app from the notification before the first notification was acknowledged, fires the notification twice which is **defiantly unwanted**.

both messages seem to be exactly the same.
{"foreground":false,"coldstart":false}
{"asleep":true,"closed":false}

IOS 9.2.1

Cordova CLI: 5.4.1
Gulp version:  CLI version 3.9.0
Gulp local:
Ionic Framework Version: 2.0.0-rc.0
Ionic CLI Version: 2.1.0
Ionic App Lib Version: 2.1.0-beta.1
ios-deploy version: 1.8.3
ios-sim version: 5.0.4
OS: Mac OS X El Capitan
Node Version: v6.7.0
Xcode version: Xcode 8.0 Build version 8A218a

**Comment by ericb** [comment]
Hey @sagivf 👋 

Could you describe the scenario where it pops up twice a bit more? Do you mean that you send one message, then send another, so the phone has two notifications waiting to be clicked?

**Comment by sagivf** [comment]
No, Just one message - that is opened via the the IPhone notification.

```
  this.push.rx.notification()
        .subscribe((message) => {
```

The subscribe callback function is called twice with the same message. I checked and the messages are not equal (===) just the same content.
Can you test it out on an Iphone env?

**Comment by ericb** [comment]
I'm still a bit confused as to how to try and reproduce. Can you walk me through step by step? Also, do you have a notification id for the message that pops up twice?

**Comment by sagivf** [comment]
Actually it doesn't seem to happen via the dashboard - only with my api requests.

Generally specking should a notification fire when entering the app? without tapping the notification. This doesn't seem to be the case for notification sent VIA the dashboard.

I'll try to pin it down some more.

Here is one sent VIA the api - 712f08c3-866c-4705-91ba-a021f1e3b635. (twice)
and one sent from the dashboard -  c36cbfb7-6814-49cd-bcb6-c8b0b2620b0c

You can also find me on slack (@sagiv), maybe i'll explain it better there.

**Comment by sagivf** [comment]
Again here are the steps - 
1) App is in background.
2) notification sent VIA api.
3) when notification pops from the top click it.
4) the notification callback is called twice.

**Comment by sagivf** [comment]
Request body

```
yield request.post({                             
  url: ionic_io_server + '/push/notifications',  
  headers: {                                     
    'Content-Type': 'application/json',          
    'Authorization': `Bearer ${ionic_io_token}`  
  },                                             
  json: {                                        
    'user_ids': [ studentId ],                   
    'profile': process.env.PUSH_ENV,             
    'notification': {                            
      'message': msg,                            
      'payload': payload,                        
      'android': {                               
        // "delay_while_idle": true,             
        // "collapse_key": false,                
        'collapse_key': 'spiklils',              
        'collapseKey': 'spiklils',               
        'delay_while_idle': false,               
        'delayWhileIdle': false,                 
        'timeToLive': 3000                       
      },                                         
      'ios': {                                   
        'sound': 'sound.wav',                    
        'content_available': true,               
        'expiry': 1423238641,                    
        'priority': 10,                          
        'contentAvailable': true                 
      }                                          
    }                                            
  }                                              
})                                               
```

**Comment by sagivf** [comment]
I think I may of found the problem - might be `contentAvailable: true` - i'll validate it soon, any idea what that is?

**Comment by ericb** [comment]
Ah okay, I understand now. So the `content_available` flag is used to specify background notifications. It looks like you have both `content_available` and `contentAvailable`. You should take a look at the [push create](http://docs.ionic.io/api/endpoints/push.html#post-notifications) endpoint for clarification on the key/value that should be sent for those requests.

But aside from that, I'm still not certain why that would cause another notification event to be fired. Curious to see if it happens when you remove those values.

**Comment by sagivf** [comment]
OK, I figured it out. It is indeed the content_available property.
So I thing there is a bug or a misunderstanding here.
From what I read content_available mean that the notification will not be displayed, but is it will be processed by the app. 

What happens is that it triggers once in the app, but still appears as a notification, which makes it trigger a second time when clicking the notification.

I'll close this and open a new ticket on the ionic cloud.
